### PR TITLE
feat: ban deposits interop

### LINKED ISSUE #10867: Interop: Ban Deposits that Trigger Executing Messages for Devnet
## Problem Statement
It is possible to include a deposit that has an invalid executing message. Deposits are force include and invalid executing messages are reorg'd out, so this means an attacker would be able to force reorgs on the chain. There are a [few solutions](https://github.com/ethereum-optimism/optimism/issues/10867#issuecomment-2183111697) to this problem, but we need to research them and there are higher priority things in the short term.

## Design
We want to ban deposit transactions completely from triggering executing messages to prevent this attack in the short term. We can reenable them in the future (after mainnet release via another hardfork). The design that we have come to consensus on is as follows:
- add an `isDeposit()(bool)` function to `L1Block`
- add a new function `L1Block.setL1BlockValuesInterop` that sets `isDeposit = true` then calls `setL1BlockValuesEcotone`
- add a new function `depositsComplete()` which sets `isDeposit = false`
- add a new deposit tx to derivation pipeline that calls `depositsComplete()`
- the `CrossL2Inbox` does `require(L1Block.isDeposit() == false)` 

~~It was determined in https://github.com/ethereum-optimism/optimism/issues/10870 that there was a missing edge case with deposit handling in the context of interop without the "only EOA" invariant. The semantics are being defined in https://github.com/ethereum-optimism/design-docs/pull/13 but it was decided that we should not implement this as part of the devnet release and instead just ban deposits that trigger executing messages completely.~~

~~The only way to do this reliably is by hacking into the state transition function. Since we don't have an explicit goal of being multiclient and we want to run the devnet to see how stable our software is, we can modify the STF temporarily to ban deposits and then remove this hack when we have a proper solution implemented.~~

~~An easy way to hack this solution is to hook into someplace around [applyTransaction](https://github.com/ethereum-optimism/op-geth/blob/5838916e90da70c6c28f6a80d5042156ca6e2a74/core/state_processor.go#L108) and revert deposit transactions that trigger an executing message event.~~

~~While this code is temporary and not going to live forever, it will help us in the short term be able to reliably test the end to end flow of the system on a devnet.~~

### PR DESCRIPTION
**Pull request by skeletor-spaceman**
**Description**

To get ready for Interop, we need to make sure to disable `ExecutingMessages` to be referenced by *Deposit* transactions.

This PR touches both the client and the relevant pre-deploy contracts.

more context: [ticket](https://github.com/ethereum-optimism/optimism/issues/10867) & [spec](https://github.com/ethereum-optimism/specs/pull/258)

**Tests**

Both solidity and go additions and modifications have been properly tested.

**Additional context**

you can check #11362 for more context on the conversations leading to this final PR

**Metadata**

- Fixes [#10867](https://github.com/ethereum-optimism/optimism/issues/10867)

### CODE REVIEW

File: op-node/rollup/derive/deposit_source.go [hunk 1]
```diff
@@ -13,9 +13,10 @@ type UserDepositSource struct {
 }
 
 const (
-	UserDepositSourceDomain    = 0
-	L1InfoDepositSourceDomain  = 1
-	UpgradeDepositSourceDomain = 2
+	UserDepositSourceDomain       = 0
+	L1InfoDepositSourceDomain     = 1
+	UpgradeDepositSourceDomain    = 2
+	AfterForceIncludeSourceDomain = 3
```
Comment (tynes) [inline op-node/rollup/derive/deposit_source.go:19]: Maybe I am missing something but I do not see this in the specs: https://github.com/ethereum-optimism/specs/blob/main/specs/interop/derivation.md

File: packages/contracts-bedrock/src/L2/CrossL2Inbox.sol [hunk 2]
```diff
@@ -71,7 +81,7 @@ contract CrossL2Inbox is ICrossL2Inbox, ISemver, TransientReentrancyAware {
 
     /// @notice Semantic version.
     /// @custom:semver 1.0.0-beta.5
-    string public constant version = "1.0.0-beta.5";
+    string public constant version = "1.1.0-beta.5";
```
Comment (tynes) [inline packages/contracts-bedrock/src/L2/CrossL2Inbox.sol]: I think its better to bump the beta version given this is unreleased, it the suffix `5` -> `6`

File: packages/contracts-bedrock/src/L2/L1BlockIsthmus.sol [hunk 3]
```diff
@@ -19,9 +20,9 @@ enum ConfigType {
 
 /// @custom:proxied
 /// @custom:predeploy 0x4200000000000000000000000000000000000015
-/// @title L1BlockInterop
-/// @notice Interop extenstions of L1Block.
-contract L1BlockInterop is L1Block {
+/// @title L1BlockIsthmus
```
Comment (tynes) [inline packages/contracts-bedrock/src/L2/L1BlockIsthmus.sol:23]: The smart contract diff looks good to me, It could be useful to submit the contracts as their own PR to make the review smaller

**Comment by protolambda** [comment]
Rebased on develop, applied a bunch of fixes/improvements to the tests, fixed the deposit-source of the comple-deposits tx not mixing in the sequence-number properly, and updated metadata etc.

Now just need the spec to properly document the new deposit-source, and then ready to merge.

File: op-node/rollup/derive/deposit_source.go (hunk 1 above)
Comment (tynes) [reply op-node/rollup/derive/deposit_source.go:19]: https://github.com/ethereum-optimism/specs/pull/370 closes this

**Comment by tynes** [comment]
Looks like the contracts failures come from warnings from solc

```
Compiler run failed:
Warning (9302): Return value of low-level calls not used.
   --> test/BenchmarkTest.t.sol:239:9:
    |
239 |         address(l1Block).call(abi.encodePacked(setValuesEcotoneCalldata));
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Perhaps we should use the `SafeCall` library to not copy back the calldata to keep gas estimation realistic and fix this error

**Comment by protolambda** [comment]
Rebasing this PR on develop, and updating the interfaces metadata

**Comment by protolambda** [comment]
Rebased. But `just interfaces-check-no-build` complains about the solidity version that is used in `CrossL2Inbox` (affecting the `IL1Block` interface). What solidity version should it be on? cc @tynes.

**Review by tynes — APPROVED** [review]

**Review by protolambda — APPROVED** [review]
